Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.cpp

```diff
@@ -461,7 +463,9 @@ void registerWebKitGStreamerVideoEncoder()
         GStreamerRegistryScanner::singleton().refresh();
 }
 
-static Lock s_activePipelinesMapLock;
+// We use a recursive lock because the removal of a pipeline can trigger the removal of another one,
+// from the same thread, specially when using chained element harnesses.
+static RecursiveLock s_activePipelinesMapLock;
```

Comment:
I checked this class and it is interesting that the unlock does not check to be the same thread the one that unlocks before decreasing the count.